src/pip/_internal/cli/cmdoptions.py

```diff
@@ -376,38 +389,41 @@ def trusted_host():
 
 def constraints():
     # type: () -> Option
-    return Option(
+    return _PathOption(
         '-c', '--constraint',
         dest='constraints',
         action='append',
         default=[],
         metavar='file',
+        type='path',
         help='Constrain versions using the given constraints file. '
         'This option can be used multiple times.'
     )
 
 
 def requirements():
     # type: () -> Option
-    return Option(
+    return _PathOption(
         '-r', '--requirement',
         dest='requirements',
         action='append',
         default=[],
         metavar='file',
+        type='path',
```

Comment:
This shouldn't use `_PathOption` as it can be a URL too.

Reply:
Gosh TIL